apps/src/maze/bee.js

```diff
@@ -274,6 +271,28 @@ export default class Bee extends Gatherer {
     this.nectars_.push({ row, col });
   }
 
+  // Overridable event handlers
+
+  onNotAtFlower = () => {};
+  setNotAtFlowerHandler(handler) {
+    this.onNotAtFlower = handler;
+  }
+
+  onFlowerEmpty = () => {};
+  setFlowerEmptyHandler(handler) {
+    this.onFlowerEmpty = handler;
+  }
+
+  onNotAtHive = () => {};
+  setNotAtHiveHandler(handler) {
+    this.onNotAtHive = handler;
+  }
+
+  onHiveFull = () => {};
+  setHiveFullHandler(handler) {
+    this.onHiveFull = handler;
+  }
+
```

Comment:
I'd be tempted to make these event handlers that can be registered instead of methods that can be overridden.  Like `elm.addEventListener('load', () => {...})` instead of `elm.onload = () => {...}`.

That way more than one thing can potentially bind to each event, and the consumer doesn't have to worry about stomping on another handler.

Reply:
...Which I guess begs the question: is it better to register a separate handler for each event, or one handler that dispatches events of different types?  Not sure I have a good answer here.

Reply:
I was debating going with that approach; I hadn't thought of consumers stomping on other handlers, and given that possibly I like actual events a lot more. I'll refactor!

I think I like separate handlers, at the very least because it'll be easier to combine them later on if we decide we want just one than it would be to separate them out if we go the opposite route.